homeassistant/components/openweathermap/weather_update_coordinator.py

```diff
@@ -205,9 +209,21 @@ def _calc_precipitation(rain, snow):
         if WeatherUpdateCoordinator._get_snow(snow) != "not snowing":
             snow_value = WeatherUpdateCoordinator._get_snow(snow)
 
-        if round(rain_value + snow_value, 1) == 0:
+        if round(rain_value + snow_value, 2) == 0:
             return None
-        return round(rain_value + snow_value, 1)
+        return round(rain_value + snow_value, 2)
+
+    @staticmethod
+    def _calc_precipitation_kind(rain, snow):
+        """Determine the precipitation kind."""
+        if WeatherUpdateCoordinator._get_rain(rain) != "not raining":
+            if WeatherUpdateCoordinator._get_snow(snow) != "not snowing":
+                return "sleet"
+            return "rain"
+
+        if WeatherUpdateCoordinator._get_snow(snow) != "not snowing":
+            return "snow"
+        return "none"
```

Comment:
Could this just return a `None` value instead of the string `"none"`?

Reply:
It probably would. I'm trying to stick with what the integration seems to be doing in other places, and not be too disruptive, so a string seemed a good choice.

Reply:
Yeah, I realized my below comments were not actually on code you changed. But, since you're doing an improvement already, might not hurt to clean it up a bit.

Reply:
You're right, a different pull request would make sense for this.